src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -23,6 +25,28 @@ open Microsoft.Quantum.QsCompiler.Transformations.QsCodeOutput
 
 // utils for verifying types in expressions
 
+/// The context used for symbol resolution and type checking.
+type ResolutionContext<'a> =
```

Comment:
Given that this is used also for verifying statements, it it a bit odd to me that it would be found within the Expression module and it looks pretty much like a nested class from within C#. 
I would instead suggest rename the namespace in SymbolTracker.fs to be just Microsoft.Quantum.QsCompiler.SyntaxProcessing and move the ResolutionContext into that file (and possibly rename that file). That seems like the better structure to me.

Reply:
I agree that that's a better place to put it. I'm not sure what to rename the file to - I want to call it `ResolutionContext.fs`, since `ResolutionContext` contains `SymbolTracker` so it feels like the more important type. But that is kind of confusing since `ContextVerification.fs`, which is a different kind of context, is right below it.

Reply:
Ah, naming... For maximal confusion you can name it VerificationContext.fs...
Maybe ResolutionState.fs? or ContextTracking.fs?

Reply:
I renamed `ResolutionContext` to `ScopeContext` and `SymbolTracker.fs` to `ScopeTools.fs`.